# This patch file was generated by NetBeans IDE
# Following Index: paths are relative to: D:\solr\lucene_solr_3_1\lucene\src\java\org\apache\lucene\search\spans
# This patch can be applied using context Tools: Patch action on respective folder.
# It uses platform neutral UTF-8 encoding and \n newlines.
# Above lines and this line are ignored by the patching process.
Index: NearSpansOrdered.java
--- NearSpansOrdered.java Base (BASE)
+++ NearSpansOrdered.java Locally Modified (Based On LOCAL)
@@ -226,14 +226,15 @@
     int start1 = spans1.start();
     int start2 = spans2.start();
     /* Do not call docSpansOrdered(int,int,int,int) to avoid invoking .end() : */
-    return (start1 == start2) ? (spans1.end() < spans2.end()) : (start1 < start2);
+    int end1 = spans1.end();
+    return (start1 == start2) ? (end1 < spans2.end()) : (start1 < start2 && end1 <= start2);
   }
 
   /** Like {@link #docSpansOrdered(Spans,Spans)}, but use the spans
    * starts and ends as parameters.
    */
   private static final boolean docSpansOrdered(int start1, int end1, int start2, int end2) {
-    return (start1 == start2) ? (end1 < end2) : (start1 < start2);
+    return (start1 == start2) ? (end1 < end2) : (start1 < start2 && end1 <= start2);
   }
 
   /** Order the subSpans within the same document by advancing all later spans
